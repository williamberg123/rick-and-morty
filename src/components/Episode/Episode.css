.Episode {
    background-color: rgb(238, 132, 12);
    padding: 10px;
    border-radius: 10px;
    font-weight: bold;
    border: 1px solid black;
}

.Episode-name {
    color: white;
}

.Episode-episode {
    color: rgb(255, 0, 0);
}

.Episode-air-date {
    color: rgb(255, 255, 255);
}