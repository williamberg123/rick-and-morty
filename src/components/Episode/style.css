.Episode {
    background-color: rgb(238, 132, 12);
    padding: 10px;
    border-radius: 10px;
    font-weight: bold;
    border: 1px solid black;
}

.Episode:hover {
    transform: scale(1.05);
    transition: 0.3s;
    border: 3px solid red;
}

.Episode-name {
    color: white;
}

.Episode-episode {
    color: rgb(255, 0, 0);
}

.Episode-air-date {
    color: rgb(255, 255, 255);
}