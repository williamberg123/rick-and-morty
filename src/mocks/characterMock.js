export const characterMock = [
    { id: 1, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 2, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 3, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 4, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 5, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 6, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 7, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 8, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 9, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 10, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 11, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 12, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 13, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 14, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 15, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 16, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 17, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 18, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 19, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
    { id: 20, name: 'Rick Sanchez', species: 'Human', image: 'img/img.png' },
];
