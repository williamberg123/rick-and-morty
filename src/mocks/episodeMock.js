export const episodeMock = {
    id: 1,
    name: 'Pilots',
    air_date: 'December 12, 21',
    episode: 'S01E01'
};
