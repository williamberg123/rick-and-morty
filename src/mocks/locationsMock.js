export const locationsMock = [
    {id: 1, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 2, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 3, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 4, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 5, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 6, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 7, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 8, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 9, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 10, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 11, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 12, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 13, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 14, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 15, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 16, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 17, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 18, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 19, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'},
    {id: 20, name: 'Earth (C-137)', type: 'Planet', dimension: 'Dimension C-137'}
];
