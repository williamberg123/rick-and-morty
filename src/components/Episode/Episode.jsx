import React from 'react';

import PropTypes from 'prop-types';

import './Episode.css';

export default function Episode({ id, name, air_date, episode }){
    const convert = () => {
        const temp = episode.slice(1, 3);
        const ep = episode.slice(4, 6);

        return `Temp: ${temp}, Ep: ${ep}`;
    };

    const episodeSeason = convert();

    return (
        <div className="Episode">
            <div className="Episode-name">#{id} - {name}</div>
            <div className="Episode-episode">{episodeSeason}</div>
            <div className="Episode-air-date">Data de exibição: {air_date}</div>
        </div>
    );
}

Episode.propTypes = {
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    air_date: PropTypes.string.isRequired,
    episode: PropTypes.string.isRequired
};