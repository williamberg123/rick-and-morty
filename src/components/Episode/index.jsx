import React, { memo } from 'react';
import { Link } from 'react-router-dom';

import PropTypes from 'prop-types';

import convert from '../../utils/convertEpisode';

import './style.css';

function Episode({ id, name, air_date, episode }){
    const episodeSeason = convert(episode);

    return (
        <Link to={`/rick-and-morty/episode?id=${id}`}>
            <div className="Episode">
                <div className="Episode-name">#{id} - {name}</div>
                <div className="Episode-episode">{episodeSeason}</div>
                <div className="Episode-air-date">Data de exibição: {air_date}</div>
            </div>
        </Link>
    );
}

Episode.propTypes = {
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    air_date: PropTypes.string.isRequired,
    episode: PropTypes.string.isRequired
};

export default memo(Episode);