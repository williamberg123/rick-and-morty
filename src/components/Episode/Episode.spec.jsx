import React from 'react';
import { render, screen } from '@testing-library/react';
import Episode from './index';

import { episodeMock } from '../../mocks/episodeMock';

describe('<Episode />', () => {
    it('should render Episode correctly', () => {
        expect.assertions(3);

        const { debug } = render(<Episode {...episodeMock} />);

        expect(screen.getByText(/Pilots/i)).toBeInTheDocument();
        expect(screen.getByText(/Temp:/i)).toBeInTheDocument();
        expect(screen.getByText(/Data de exibição:/i)).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        render(<Episode {...episodeMock} />);

        expect(screen.getByText(/Pilots/i).parentElement)
            .toMatchSnapshot();
    });
});

// Testado