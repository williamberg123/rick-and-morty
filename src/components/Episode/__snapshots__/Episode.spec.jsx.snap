// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Episode /> should match snapshot 1`] = `
<div
  class="Episode"
>
  <div
    class="Episode-name"
  >
    #
    1
     - 
    Pilots
  </div>
  <div
    class="Episode-episode"
  >
    Temp: 01, Ep: 01
  </div>
  <div
    class="Episode-air-date"
  >
    Data de exibição: 
    December 12, 21
  </div>
</div>
`;
